### PR TITLE
Update self_hosted_runner: remove deprecated docker.userEmulation and use containerOptions

### PR DESCRIPTION
**Pull request by mirpedrol**
~~Modifications to match https://github.com/nf-core/tools/pull/2673 and fix the tests there.~~

The previous PR won't be merged. But `userEmulation` is deprecated, so we should remove it.
Also remove the preceding `docker.` as all options are inside the docker scope.

### CODE REVIEW

File: conf/self_hosted_runner.config [hunk 1]
```diff
@@ -15,8 +15,6 @@ params {
 
 docker {
     enabled = true
-    docker.fixOwnership = true
-    docker.userEmulation = false
-    docker.runOptions = '--platform=linux/amd64'
+    fixOwnership = true
+    runOptions = '--platform=linux/amd64'
```
Comment (maxulysse) [inline conf/self_hosted_runner.config:19]: not sure about keeping `fixOwnership`

**Comment by jfy133** [comment]
@mirpedrol do you still plan to finish this?

**Comment by mirpedrol** [comment]
I think this can be merged, if someone could review 🙏

**Review by mashehu — APPROVED** [review]

**Review by jfy133 — APPROVED** [review]

**Comment by jfy133** [comment]
Ok approved but unable to merge from my phone... @mirpedrol does it give you the option to do so?

**Comment by mirpedrol** [comment]
Thank you!!
The pre-commit test is required but not running, did we remove this test? should I remove the requirement from the branch protection?

**Comment by mirpedrol** [comment]
The branch was not up to date 😄  I will merge it once all tests pass!

**Pull request merged**